compiler/pkg/openapi_generator/check-backward-compatibility.go

```diff
@@ -0,0 +1,142 @@
+package openapi_generator
+
+import (
+	"bytes"
+	"errors"
+	"fmt"
+	"os"
+	"path/filepath"
+	"strings"
+
+	"github.com/ghodss/yaml"
+	log "github.com/sirupsen/logrus"
+
+	nexus_compare "github.com/vmware-tanzu/graph-framework-for-microservices/common-library/pkg/nexus-compare"
+	extensionsv1 "k8s.io/apiextensions-apiserver/pkg/apis/apiextensions/v1"
+)
+
+func splitCRDs(content []byte) []string {
+	return strings.Split(string(content), "---")
+}
+
+func compareCRDs(inCompatibleCRDs []*bytes.Buffer, existingCRDName, existingCRDContent string, newCRDContent []byte) ([]*bytes.Buffer, error) {
+	newCRDParts := splitCRDs(newCRDContent)
+	for _, newCRDPart := range newCRDParts {
+		if newCRDPart == "" {
+			continue
+		}
+
+		newCRD := &extensionsv1.CustomResourceDefinition{}
+		err := yaml.Unmarshal([]byte(newCRDPart), newCRD)
+		if err != nil {
+			return nil, fmt.Errorf("error unmarshaling new CRD: %v", err)
+		}
+
+		if newCRD.Name != existingCRDName {
+			continue
+		}
+
+		// When there is a backward incompatibility, we fail the build if we don't force an upgrade.
+		isInCompatible, message, err := nexus_compare.CompareFiles([]byte(existingCRDContent), []byte(newCRDPart))
+		if err != nil {
+			panic(fmt.Sprintf("Error occurred while checking CRD's %q backward compatibility: %v", existingCRDName, err))
+		}
+		if isInCompatible {
+			log.Warnf("CRD %q is incompatible with the previous version", existingCRDName)
+			inCompatibleCRDs = append(inCompatibleCRDs, message)
+		}
+	}
+	return inCompatibleCRDs, nil
+}
+
+func CheckBackwardCompatibility(existingCRDsPath, yamlsPath string, force bool) error {
+	var (
+		removedCRDs      []string
+		inCompatibleCRDs []*bytes.Buffer
+	)
+
+	if err := filepath.Walk(existingCRDsPath, func(path string, info os.FileInfo, err error) error {
+		if err != nil {
+			if errors.Is(err, os.ErrNotExist) {
+				fmt.Printf("No files exists on the path %q: %v", existingCRDsPath, err)
```

Comment:
can we change `existingCRDsPath` to `path`, so that full path will be seen in the error